### ISSUE TITLE
Question about UA mobile phone number validation.

### ISSUE THREAD

**Issue by EugeneEl** [marmelroy/PhoneNumberKit#656]
Hello, thank you for your great work on this project.

I have a question about parsing/validation UA phone numbers.
I don't understand why this number considered as valid: `3805011111`.

```swift
import Foundation
import PhoneNumberKit

class PhoneNumberValidationUtils {
    
    private static let phoneNumberKit = PhoneNumberKit()
    
    static func isPhoneNumberValid(_ phoneNumber: String) -> Bool {
        let clearedPhone = phoneNumber.replacingOccurrences(of:"[^0-9]", with: "", options: .regularExpression)
        let isValid = phoneNumberKit.isValidPhoneNumber(clearedPhone)
        print("phoneNumber \(clearedPhone) is valid: \(isValid)")
        return isValid
//        do {
//            print("valid phone number: \(phoneNumber)")
//            _ = try phoneNumberKit.isValidPhoneNumber(phoneNumber)
//            return true
//        }
//        catch {
//            print("invalid phone number: \(phoneNumber)")
//            return false
//        }
    }
} 
```

```Output
phoneNumber 3805011111 is valid: true
```

I guess UA phone number should contain 7 digits after carrier code.
But I have just 5 and it valid. And `38050111111` is not valid.

I checked [here](https://htmlpreview.github.io/?https://github.com/google/libphonenumber/blob/master/javascript/i18n/phonenumbers/demo-compiled.html) and got the another results.

```
****Parsing Result:****
{"country_code":380,"national_number":3805011111,"raw_input":"3805011111","country_code_source":20}

****Validation Results:****
Result from isPossibleNumber(): true
Result from isValidNumber(): false
Phone Number region: UA
Result from getNumberType(): UNKNOWN

****ShortNumberInfo Results:****
Result from isPossibleShortNumber: false
Result from isValidShortNumber: false
Result from isPossibleShortNumberForRegion: false
Result from isValidShortNumberForRegion: false

****Formatting Results:**** 
E164 format: invalid
Original format: 3805011111
National format: 3805011111
International format: invalid
Out-of-country format from US: invalid
Out-of-country format from Switzerland: invalid
Format for mobile dialing (calling from US): invalid
Format for national dialing with preferred carrier code and empty fallback carrier code: invalid

****AsYouTypeFormatter Results****
Char entered: 3 Output: 3
Char entered: 8 Output: 38
```

Of course my question is more related to phone number validation itself, but it would be cool if you would find some time to help.

Also is it possible to customize validation rules like add min required length for specific country?

**Comment by EugeneEl** [comment]
Also is it possible to check that phone number is in international format?

**Comment by bguidolim** [comment]
I guess you're not using the latest version.

**Comment by EugeneEl** [comment]
```ruby
    # Swift 5.3 framework for parsing, formatting and validating international phone numbers. Inspired by Google's libphonenumber.
    # https://github.com/marmelroy/PhoneNumberKit
    pod 'PhoneNumberKit', :git => 'https://github.com/marmelroy/PhoneNumberKit.git', :commit => '6edd6e38a30aec087cb97f7377edf876c29a427e'
```

I'm using this version, as I understand fix for iOS 16.4 hasn't been released yet to CocoaPods.

```swift
import Foundation
import PhoneNumberKit

class PhoneNumberValidationUtils {
    
    private static let phoneNumberKit = PhoneNumberKit()
    
    static func isPhoneNumberValid(_ phoneNumber: String) -> Bool {
        let clearedPhone = phoneNumber.replacingOccurrences(of:"[^0-9]", with: "", options: .regularExpression)
        let isValid = phoneNumberKit.isValidPhoneNumber(clearedPhone)
        
        if !isValid {
            return false
        }
        
        do {
            let number = try phoneNumberKit.parse(clearedPhone)
            if number.type == .mobile {
                print("valid phone number: \(clearedPhone)")
            } else {
                print("invalid phone number: \(clearedPhone)")
            }
            return number.type == .mobile
        }
        catch {
            print("invalid phone number: \(phoneNumber)")
            return false
        }
    }
}
```

I need to check that entered phone is both valid and mobile. Tried this approach. 
Is it correct way? Not sure I understand the difference between `parse` and `isValid` functions
